0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+|:---------|:---------------------|:------------|:-----------------------------|
+| Transfer | `id`                 | GET         | [Get Transfer][]             |
+| Transfer | `ledger`             | GET         | [Get Ledger Metadata][]      |
+| Transfer | `fulfillment`        | PUT         | [Submit Fulfillment][]       |
+| Transfer | `fulfillment`        | GET         | [Get Transfer Fulfillment][] |
+| Transfer | `transfer_rejection` | PUT         | [Reject Transfer][]          |
+| Transfer | `debit_account`      | GET         | [Get Account][]              |
+| Transfer | `credit_account`     | GET         | [Get Account][]              |
+| Message  | `from`               | GET         | [Get Account][]              |
+| Message  | `to`                 | GET         | [Get Account][]              |
+| Message  | `ledger`             | GET         | [Get Ledger Metadata][]      |
+
+
+### Date-Time
+[Date-Time]: #date-time
+
+All dates and times should be expressed in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date-time format with precision to the second or millisecond. The time zone offset should always be `Z`, for no offset. In other words, the date-time should be a **string** matching the one of the following formats:
+
+| Precision   | Format                     |
+|:------------|:---------------------------|
+| Second      | `YYYY-MM-DDTHH:mm:ss.sssZ` |
+| Millisecond | `YYYY-MM-DDTHH:mm:ssZ`     |
+
+
+## API Methods
+
+### Get Ledger Metadata
+[Get Ledger Metadata]: #get-server-metadata
+
+Receive information about the ledger.
+
+**Authorization:** This method MUST be accessible to all clients authorized as account owners or administrators. This method MAY be accessible to unauthorized users.
+
+#### Request Format
+
+```
+GET /
+```
+
+**Metadata URL Name:** None (top-level path for the ledger)
+
+##### Response Format
+
+A successful result uses the HTTP response code **200 OK** and contains a JSON object with the following fields. (A field name like `foo[].bar` indicates that the row describes the field `bar` in _each_ object contained in the array `foo`.)
+
+| Field               | Value              | Description                       |
+|:--------------------|:-------------------|:----------------------------------|
+| `asset_info`        | [Asset resource][] | Information on the currency or other asset this ledger tracks. |
+| `ilp_prefix`        | String             | The ILP Address prefix of the ledger. This must match the regular expression `^[a-zA-Z0-9._~-]+\.$` |
+| `connectors`        | Array of Objects   | Accounts belonging to recommended ILP Connectors. This list MAY be empty. |
+| `connectors[].id`   | [URL][]            | The unique ID of this connector's account. This is used when [messaging](#send-message) the connectors to get quotes. MUST be an HTTP(S) URL where a client can [Get Account][]. |
+| `connectors[].name` | [URL][]            | The unique `name` field of this connector's account. |
```

Comment:
Right now ilp-core uses `name` and ignores `id` (see https://github.com/interledgerjs/ilp-core/blob/11a019344b254d33d8cddb7ddfef0642d47977f2/src/lib/client.js#L227-L237).